src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -391,6 +401,75 @@ protected virtual void Initialize()
             }
         }
 
+        // Scan the extensions directory and Load custom extension.
+        // $$$ - need to finalize this policy. Currently enabled via an AppSettting. 
+        private void LoadCustomExtensions(JObject hostConfig)
+        {
+            var bindingRoot = ConfigurationManager.AppSettings["BYOB_Path"];
+            if (!string.IsNullOrWhiteSpace(bindingRoot))
+            {
+
+                var config = this.ScriptConfig.HostConfig;
+
+                foreach (var dir in Directory.EnumerateDirectories(bindingRoot))
+                {
+                    foreach (var path in Directory.EnumerateFiles(dir, "*.dll"))
+                    {
+                        // How to find the extension assembly? Don't want to load every one to inspect it. 
+                        // $$$ manifest file? scan for any assembly with 
+                        if (!path.ToLowerInvariant().Contains("extension"))
+                        {
+                            continue;
+                        }
+
+                        // See GetNugetPackagesPath() for details 
+                        // Script runtime is already setup with assembly resolution hooks, so use LoadFrom
+                        Assembly assembly = Assembly.LoadFrom(path);
+                        LoadExtensions(assembly, path, config, hostConfig);
+                    }
+                }
+
+                // Get tooling after all extensions have been initialized. 
+                var generalProvider = ScriptConfig.BindingProviders.OfType<GeneralScriptBindingProvider>().First();
+                generalProvider.Tooling = config.GetTooling();
+            }
+        }
+
+        // return true if any extensions loaded, else false. 
+        private void LoadExtensions(Assembly assembly, string locationHint, JobHostConfiguration config, JObject hostConfigMetadata)
+        {
+            foreach (var type in assembly.ExportedTypes)
+            {
+                if (!typeof(IExtensionConfigProvider).IsAssignableFrom(type))
+                {
+                    continue;
+                }
+
+                try
+                {
+                    IExtensionConfigProvider instance = (IExtensionConfigProvider)Activator.CreateInstance(type);
+
+                    // Apply config to the extension object. 
+                    JsonConvert.PopulateObject(hostConfigMetadata.ToString(), instance);
+
+                    string name = type.Name;
+
+                    var httpHook = instance as IAsyncConverter<HttpRequestMessage, HttpResponseMessage>;
+                    if (httpHook != null)
+                    {
+                        this._customHttpHandlers[name] = httpHook;
```

Comment:
No, this can't be handled with custom receivers.  Basically, the extensions we're enabling (notably Graph) need to have an http point open and that may do internal logic and dispatch multiple user-level triggers.